jetbrains-core/src/software/aws/toolkits/jetbrains/core/explorer/ExplorerToolWindow.kt

```diff
@@ -64,8 +68,15 @@ class ExplorerToolWindow(private val project: Project) : SimpleToolWindowPanel(t
     }
 
     init {
+        val group = DefaultActionGroup(
+            SettingsSelectorComboBoxAction(project, ChangeAccountSettingsMode.CREDENTIALS),
+            SettingsSelectorComboBoxAction(project, ChangeAccountSettingsMode.REGIONS)
```

Comment:
I'm curious how this works, here we use credentials and regions but we show both below? It seems like the enum is doing too many things or I'm just confused on how it should be used?

Reply:
You're *probably* right the enum might be doing too many things, but the logic is all encapsulated together - found myself doing a bunch of 

kt
```
when(mode) {
  BOTH -> logic
  REGIONS -> logic
}
```

stuff- and figured we could collapse it down.